lang/guile2/Portfile

```diff
@@ -0,0 +1,172 @@
+    supported_archs i386 x86_64
+    set universal_archs_supported {i386 x86_64}
+} elseif {${os.arch} eq "i386" && ${build_arch} eq "x86_64"} {
+    supported_archs i386 x86_64 ppc
+    set universal_archs_supported {i386 x86_64 ppc}
+} elseif {${os.arch} eq "i386"} {
+    supported_archs i386 ppc
+    set universal_archs_supported {i386 ppc}
+} elseif {${build_arch} eq "ppc64"} {
+    supported_archs ppc ppc64
+    set universal_archs_supported {ppc ppc64}
+} else {
+    supported_archs ${build_arch}
+    set universal_archs_supported ${build_arch}
+}
+
+platform darwin {
+#    if {[variant_isset universal]} {
+#        set merger_host(x86_64) x86_64-apple-${os.platform}${os.major}
+#        set merger_host(i386) i686-apple-${os.platform}${os.major}
+#        set merger_host(ppc64) powerpc64-apple-${os.platform}${os.major}
+#        set merger_host(ppc) powerpc-apple-${os.platform}${os.major}
+#        set merger_configure_args(x86_64) "--build=x86_64-apple-${os.platform}${os.major}"
+#        set merger_configure_args(i386) "--build=i686-apple-${os.platform}${os.major}"
+#        set merger_configure_args(ppc) "--build=powerpc-apple-${os.platform}${os.major}"
+#        set merger_configure_args(ppc64) "--build=powerpc64-apple-${os.platform}${os.major}"
+#    } else
+    if {${build_arch} eq "i386"} {
+        configure.args-append \
+            --host=i686-apple-${os.platform}${os.major} \
+            --build=i686-apple-${os.platform}${os.major}
+    } elseif {${build_arch} eq "x86_64"} {
+        configure.args-append \
+            --host=x86_64-apple-${os.platform}${os.major} \
+            --build=x86_64-apple-${os.platform}${os.major}
+    } elseif {${build_arch} eq "ppc"} {
+        configure.args-append \
+            --host=powerpc-apple-${os.platform}${os.major} \
+            --build=powerpc-apple-${os.platform}${os.major}
+    } elseif {${build_arch} eq "ppc64"} {
+        configure.args-append \
+            --host=powerpc64-apple-${os.platform}${os.major} \
+            --build=powerpc64-apple-${os.platform}${os.major}
+    }
+}
+
+post-destroot {
+    # Remove conflicting files and rename to make guile2 installable
+    # with guile 3.0. Update internal paths and links.
+    file delete ${destroot}${prefix}/share/info/guile.info
+    file delete ${destroot}${prefix}/share/info/guile.info-1
+    file delete ${destroot}${prefix}/share/info/guile.info-2
+    file delete ${destroot}${prefix}/share/info/guile.info-3
+    file delete ${destroot}${prefix}/share/info/guile.info-4
+    file delete ${destroot}${prefix}/share/info/guile.info-5
+    file delete ${destroot}${prefix}/share/info/guile.info-6
+    file delete ${destroot}${prefix}/share/info/guile.info-7
+    file delete ${destroot}${prefix}/share/info/guile.info-8
+    file delete ${destroot}${prefix}/share/info/guile.info-9
+    file delete ${destroot}${prefix}/share/info/guile.info-10
+    file delete ${destroot}${prefix}/share/info/r5rs.info
```

Comment:
Rather than the copy-pasta, define all of these via a list, and then loop through them. Ditto for the file renames.

Reply:
Checking if deleting works first. Have to rely on CI here, rebuilding `guile` takes about 10 hrs or more…

Reply:
> Checking if deleting works first. Have to rely on CI here, rebuilding `guile` takes about 10 hrs or more…

One tip for these situations: Build a test stub port - with logic like this, that needs to be validated - so you don't have to depend on CI.

I'll post an example shortly.

Reply:
Alternatively, an even easier approach is to test the logic directly in a TCL script. (While the various port-specific things like phases, etc, don't exist. But logic like looping, file deletes/renames, etc, all do. And you can simulate many things, by )

Here's an example I created back two years ago, when I was pre-testing logic for `openmpi`/`mpich`:

<details>
  <summary>mpi-testing.tcl</summary>
  
  ```tcl
#!/opt/local/bin/tclsh

if { ${argc} != 3 } {
    puts "Usage: ${argv0} <os.major> <os.arch> <debug-yes/no>"
    exit 1
}

# Define 'ui_debug', which is otherwise already available within a portfile
proc ui_debug {p_msg} {
    puts ${p_msg}
}

set os.major \
    [lindex $argv 0]
set os.arch \
    [lindex $argv 1]
set port_debug \
    [lindex $argv 2]

ui_debug "os_major: ${os.major}"
ui_debug "build_arch: ${os.arch}"
ui_debug "debug: ${port_debug}"

if { ![string is boolean -strict ${port_debug}] } {
    ui_debug "Error: arg 'debug' must be a boolean"
    exit 1
}

# Define 'variant_isset', which is otherwise already available within a portfile
proc variant_isset {p_debug} {
    global port_debug
    return ${port_debug}
}

proc mpiutil_add_subports {clist clist_unsupported clist_obsolete} {
    foreach key [dict keys ${clist}] {
        ui_debug "mpiutil_add_subport - clist - ${key}"
    }

    foreach key ${clist_unsupported} {
        ui_debug "mpiutil_add_subport - clist_unsupported - ${key}"
    }

    foreach key ${clist_obsolete} {
        ui_debug "mpiutil_add_subport - clist_obsolete - ${key}"
    }
}

set clist [dict create]
set clist_unsupported [list]
set clist_obsolete [list]

# TODO: Remove these obsolete subports on or after January 2022, along with all
# traces of -devel. For the latter, also remember to update proc
# 'mpiutil_add_subport', in portgroup 'mpiutil'.
lappend clist_obsolete \
    clang33 clang34 clang37 \
    clang50 clang60 clang70 clang80 \
    gcc49 gcc5 gcc6 gcc8

if { ${os.arch} eq "arm" } {
    # disable unreliable clang versions for ARM
    dict set clist gcc11   {macports-gcc-11}
    dict set clist clang11 {macports-clang-11}
    lappend clist_unsupported \
        clang90 clang10
} else {
    dict set clist clang90 {macports-clang-9.0}
    dict set clist clang10 {macports-clang-10}
    dict set clist clang11 {macports-clang-11}
}

# gcc 9+ only available on macOS10.7 (Darwin11) and newer
# However, gcc9+ builds fail on MacOS 10.7, so only enable for MacOS 10.8+.
if { ${os.arch} eq "arm" } {
    dict set clist gcc11 {macports-gcc-11}
    lappend clist_unsupported \
        gcc9 gcc10
} elseif { ${os.major} >= 12 } {
    dict set clist gcc9  {macports-gcc-9}
    dict set clist gcc10 {macports-gcc-10}
    dict set clist gcc11 {macports-gcc-11}
} else {
    lappend clist_unsupported \
        gcc9 gcc10 gcc11
}

ui_debug "clist: ${clist}"

mpiutil_add_subports \
    ${clist} ${clist_unsupported} ${clist_obsolete}
  ```
</details>